contracts/staking/RewardDelegators.sol

```diff
@@ -224,6 +225,7 @@ contract RewardDelegators is
     ) internal returns(uint256 _aggregateReward) {
         _updateRewards(_cluster);
 
+
         for(uint256 i = 0; i < _tokens.length; i++) {
```

Comment:
This tokens array might not always contain both POND and MPond addresses which means the tree might be updated with random values.